stl/inc/vector

```diff
@@ -1761,15 +1635,94 @@ private:
         _My_data._Orphan_all();
 
         if (_Myfirst) { // destroy and deallocate old array
-            _Destroy(_Myfirst, _Mylast);
-            _Getal().deallocate(_Myfirst, static_cast<size_type>(_Myend - _Myfirst));
+            _Destroy_range(_Myfirst, _Mylast, _Al);
+            _Al.deallocate(_Myfirst, static_cast<size_type>(_Myend - _Myfirst));
 
             _Myfirst = nullptr;
             _Mylast  = nullptr;
             _Myend   = nullptr;
         }
     }
 
+    template <class... _Valty>
+    _CONSTEXPR20_CONTAINER void _Construct_n(_CRT_GUARDOVERFLOW const size_type _Count, _Valty&&... _Val) {

```

Comment:
I'd like a comment going into the expected usage of `_Construct_n`, and detailing the motivation for the `if constexpr(sizeof...(_Val))` structure. It's not _bad_ but without some kind of comment it's very hard to know what the usage is supposed to be.

It may also be that keeping the inner part of this function as an overload set would be simpler.

Reply:
urgh, i tried something and it is terrible. any ideas how to formulate this?

Reply:
You could split it out to four functions (three in the overload set + `_Construct_n`) and do the proxy construction in `_Construct_n` before calling the overload set.